code_challenges/code_wars/trigrams.py

```diff
@@ -0,0 +1,18 @@
+def trigrams(phrase):
+    i = 0
+    j = 3
+    trigrams = []
+    updated_phrase = phrase.replace(" ", "_")
+    while i < len(phrase):
+        for character in updated_phrase:
+            print(i, j, updated_phrase[i:j])
+            trigrams.append(updated_phrase[i:j])
+            i += 1
+            j += 1
+
+    return " ".join(trigrams[:-2])
```

Comment:
What if you were to move this slice to line 7, so you never do the work to begin with rather than throwing it away at the end? 

Also you don't use `character` from line 7 so what if you changed this to a range and made it output the correct `i` value.  You then wouldn't need to increment `i`.

If i and j are always the same distance apart do you need to increment a j to begin with or could you use i for both?